docs/contributing/testing.md

```diff
@@ -23,9 +23,11 @@ Fix any CSS or JavaScript style errors that the linting tests report.
 
 To check whether your HTML markup is valid, [check your page with W3C's markup validator](https://validator.w3.org/).
 
-If you can, fix any errors or warnings the validator reports.
+You can also [check your page with W3C's 'Check serialized DOM of current page’ bookmarklet](https://validator.w3.org/nu/about.html#extras). For example, if your page has a lot of JavaScript, or if you can only check your page when you've logged in to a site. 
```

Comment:
We could try to add something like the following to make it clear what the difference is between running the above validator on a URL and using this tool:

> To check whether your HTML markup is valid, [check your page with W3C's markup validator](https://validator.w3.org/). This tool checks the source code of your document (the code you see when you select [View source](https://developer.mozilla.org/en-US/docs/Tools/View_source) in your browser). 
>
> To check whether your HTML markup is valid after JavaScript has modified it, you can also [check your page with W3C's 'Check serialized DOM of current page’ bookmarklet](https://validator.w3.org/nu/about.html#extras). You should do this if your page has a lot of JavaScript, or if you can only check your page when you've logged in to a site.

Reply:
@hannalaakso Looks great! Have just suggested slight rejig in the most recent commit. Can I get a final review before I send this PR for 2i? :)

Reply:
Thanks @EoinShaughnessy. Should we link to some instructions on how to install a bookmarklet? I've just realised that the the page we link to doesn't seem to mention how to install the checker? 🤔  Or do we think users will know how to do this?

(Alternatively, you can use the W3C markup validator to do the same by [opening the browser inspector](https://www.gov.uk/government/publications/doing-a-basic-accessibility-check-if-you-cant-do-a-detailed-one/doing-a-basic-accessibility-check-if-you-cant-do-a-detailed-one#html-checks) and copying all the page HTML to https://validator.w3.org/#validate_by_input. But that's not very easy to explain either.)

Reply:
Thanks, @hannalaakso - yeah, maybe we can move ahead with what we have now and iterate later if requested. I guess this content is for technically knowledgeable users anyway, so until corrected maybe we can assume the current signposting is enough.